src/pages/SettingsPage.js

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import {
+    View,
+    TouchableOpacity,
+} from 'react-native';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+import {withOnyx} from 'react-native-onyx';
+import {redirect} from '../libs/actions/App';
+import styles from '../styles/styles';
+import Text from '../components/Text';
+import {signOut} from '../libs/actions/Session';
+import ONYXKEYS from '../ONYXKEYS';
+import ROUTES from '../ROUTES';
+import HeaderWithCloseButton from '../components/HeaderWithCloseButton';
+import {version} from '../../package.json';
+import AvatarWithIndicator from '../components/AvatarWithIndicator';
+
+const propTypes = {
+    /* Onyx Props */
+    // The personal details of the person who is logged in
+    myPersonalDetails: PropTypes.shape({
+    // Display name of the current user from their personal details
+        displayName: PropTypes.string,
+
+        // Avatar URL of the current user from their personal details
+        avatarURL: PropTypes.string,
+    }),
+
+    // Information about the network
+    network: PropTypes.shape({
+    // Is the network currently offline or not
+        isOffline: PropTypes.bool,
+    }),
+
+    // Currently viewed reportID
+    currentlyViewedReportID: PropTypes.string,
+
+    // Information about the current session (authToken, accountID, email, loading, error)
+    session: PropTypes.shape,
+};
+
+const defaultProps = {
+    myPersonalDetails: {},
+    network: null,
+    currentlyViewedReportID: '',
+    session: {},
+};
+const SettingsPage = ({
+    myPersonalDetails, network, session, currentlyViewedReportID,
+}) => {
+    // On the very first sign in or after clearing storage these
+    // details will not be present on the first render so we'll just
+    // return nothing for now.
+    if (!myPersonalDetails || _.isEmpty(myPersonalDetails)) {
+        return null;
+    }
+    return (
+        <View
+            pointerEvents="box-none"
+            style={[
+                styles.settingsPageBackground,
+            ]}
+        >
+            <View
+                style={[
+                    styles.flexColumn,
+                    styles.settingsPageContainer,
+                ]}
+            >
+                <HeaderWithCloseButton
+                    onCloseButtonPress={() => redirect(ROUTES.getReportRoute(currentlyViewedReportID))}
```

Comment:
If `currentlyViewedReport` is `''`, where will this redirect the user?

Reply:
Heh that's a good question I think it will take you nowhere. But we should theoretically always have a reportID. Unless we are navigating straight to `#/settings` without following the usual path of moving from the homepage to somewhere else. Which I guess could happen. That said, I'm not sure we need to worry about this right now.

Reply:
An alternative for now could be to just redirect to the root if there's no `currentlyViewedReportID`.

Reply:
it goes to a blank page, do we need to redirect it to a default or just skip this for now?

Reply:
something like 

```js
currentlyViewedReportID 
    ? redirect(ROUTES.getReportRoute(currentlyViewedReportID)
    : redirect(ROUTES.HOME);
```

Reply:
I think the suggestion here should work. Most of the time there will be a reportID.